university-of-south-australia-2017-harvard.csl

```diff
@@ -283,8 +281,14 @@
           <date-part name="month" form="long" suffix=" "/>
           <date-part name="year" form="long"/>
         </date>
-        <text variable="URL" prefix=", &lt;" suffix="&gt;"/>
       </if>
+	</choose>
+  </macro>
+  <macro name="access-link">
+	<choose>
+	  <if type="article-journal book chapter" match="none">
```

Comment:
Is this right, though? On https://www.library.unisa.edu.au/referencing-roadmap/electronic/e-book/ I'm seeing

> Author's family name, Initial(s) OR Authoring body year, Title of ebook, edition if applicable, Publisher, Place of publication if available, <URL>.

and a book with a URL is by definition an e-book.

Reply:
As you can see on the style page you referenced, our style has multiple rules for displaying e-book URLs. Freely available e-books show the URL, but purchased or library e-books do not (and similarly for journals). Because we can't capture this nuance within the bounds of CSL, we've decided that as the majority of e-book and journal content available in our institution is purchased/subscribed content, the default should be to not display a URL for these content types.